src/parse/cast.jl

```diff
@@ -129,23 +129,23 @@ function parse_kwargs(def)
 end
 
 function to_argument(def, ex)
-    # (name, type, require, vararg)
+    # (name, type, require, vararg, default_value)
     @smatch ex begin
-        ::Symbol => (string(ex), Any, true, false)
-        :($name::$type) => (string(name), wrap_type(def, type), true, false)
-        :($name::$type...) => (string(name), wrap_type(def, type), true, true)
-        Expr(:kw, :($name::$type), _) => (string(name), wrap_type(def, type), false, false)
-        :($name...) => (string(name), Any, true, true)
-        Expr(:kw, name::Symbol, _) => (string(name), Any, false, false)
+        ::Symbol => (string(ex), Any, true, false, nothing)
```

Comment:
I think we need to use `Symbol` for the name if we are recording the default value. Given potentially we will want to replicate the julia function behaviour by inferring variable dependencies in the default value. So things like `f(x; y=sin(x))` will work.

Reply:
but need to check other places when changing this.